src/main/java/gov/nist/csd/pm/pap/PAPGraph.java

```diff
@@ -429,8 +431,10 @@ static void checkAccessRightsValid(Graph graph, AccessRightSet accessRightSet) t
 
     @Override
     public void dissociate(String ua, String target) throws PMException {
-        if ((!nodeExists(ua) || !nodeExists(target))
-                || (!getAssociationsWithSource(ua).contains(new Association(ua, target)))) {
+
```

Comment:
This looks good but the logic is slightly different. We don't want to call `getAssociationsWithSource` if the `ua` node does not exist. The previous logic relied on the if statement short circuiting if the node did not exist so `getAssociationsWithSource` was never called. You could change it to something like this:

```
boolean nodesNotExist = (!nodeExists(ua) || !nodeExists(target));
if (nodesNotExist) {
    return;
}

boolean pathNotExist = (!getAssociationsWithSource(ua).contains(new Association(ua, target)));
if (pathNotExist) {
    return;
}
```

to get the same logic as the short circuit.